colossalai_platform/cli/api/api.py

```diff
@@ -1,40 +1,128 @@
 import json
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, IO
 import requests
-from pydantic import BaseModel
+import enum
 
 from colossalai_platform.cli.config import Config
 
+LOGGER = logging.getLogger(__name__)
 
-class LoginError(Exception):
+
+class ApiError(Exception):
     pass
 
 
-class ColossalPlatformApi(BaseModel):
+class LoginRequiredError(Exception):
+
+    def __init__(self, message="Login required"):
+        super().__init__(message)
+
+
+class StorageType(enum.Enum):
+    DATASET = "dataset"
+    PROJECT = "project"
+
+
+@dataclass
+class ColossalPlatformApi:
     config: Config
     token: str = ""
+    session: requests.Session = requests.Session()
 
     def login(self):
-        url = "https://luchentech.com//api/user/login"
+        # TODO: a better URL construction method,
+        #       with pydantic.HttpUrl
+        url = str(self.config.api_server) + "/api/user/login"
 
         if self.config.username == "" or self.config.password == "":
-            raise LoginError("Username or password is empty, please call `cap configure` first")
+            raise ApiError("Username or password is empty, please call `cap configure` first")
 
+        headers = {'Content-Type': 'application/json'}
         payload = json.dumps({
             "username": self.config.username,
             "password": self.config.password,
         })
 
-        headers = {'Content-Type': 'application/json'}
-
-        response = requests.request(
+        response = self.session.request(
             "POST",
             url,
             headers=headers,
             data=payload,
-            verify=False,    # FIXME(ofey404): remove this
         )
 
         if response.status_code == 200:
             self.token = response.json()['accessToken']
         else:
-            raise LoginError(f"Login failed with status code {response.status_code}, body: {response.text}")
+            raise ApiError(f"Login failed with status code {response.status_code}, body: {response.text}")
+
+    def upload(
+        self,
+        storage_type: StorageType,
+        storage_id: str,
+        storage_path: str,
+        local_file_path: Path | str,
+    ):
+        # TODO(ofey404): multi part upload for large file
+        LOGGER.debug(f"Uploading {local_file_path} to {storage_type.value}://{storage_id}/{storage_path}")
+        url = self._get_presigned_upload_urls(
+            storage_type=storage_type,
+            storage_id=storage_id,
+            file_path=storage_path,
+            total_parts=1,
```

Comment:
if total_parts is one, we don't need to call complete multipart upload api?

Reply:
Yes, I've tried.

The frontend logic is the same, I've checked.